cpp/scenario/gazebo/include/scenario/gazebo/Link.h

```diff
@@ -0,0 +1,124 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SCENARIO_GAZEBO_LINK_H
+#define SCENARIO_GAZEBO_LINK_H
+
+#include <ignition/gazebo/Entity.hh>
+#include <ignition/gazebo/EntityComponentManager.hh>
+#include <ignition/gazebo/EventManager.hh>
+
+#include <array>
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace scenario {
+    namespace base {
+        struct Pose;
+        struct ContactData;
+    } // namespace base
+    namespace gazebo {
+        class Link;
+        class Model;
+    } // namespace gazebo
+} // namespace scenario
+
+class scenario::gazebo::Link
+{
+public:
+    Link();
+    virtual ~Link();
+
+    // ===========
+    // Gazebo Link
+    // ===========
+
+    uint64_t id() const;
+    bool initialize(const ignition::gazebo::Entity linkEntity,
+                    ignition::gazebo::EntityComponentManager* ecm,
+                    ignition::gazebo::EventManager* eventManager);
+    bool createECMResources();
+
+    // =========
+    // Link Core
+    // =========
+
+    std::string name() const;
+    std::array<double, 3> position() const;
+    std::array<double, 4> orientation() const;
+
+    std::array<double, 3> worldLinearVelocity() const;
+    std::array<double, 3> worldAngularVelocity() const;
+    std::array<double, 3> bodyLinearVelocity() const;
+    std::array<double, 3> bodyAngularVelocity() const;
+
+    std::array<double, 3> worldLinearAcceleration() const;
+    std::array<double, 3> worldAngularAcceleration() const;
+    std::array<double, 3> bodyLinearAcceleration() const;
+    std::array<double, 3> bodyAngularAcceleration() const;
+
+    bool contactsEnabled() const;
+    bool enableContactDetection(const bool enable);
+
+    bool inContact() const;
+    std::vector<scenario::base::ContactData> contactData() const;
+
+    bool applyWorldForce(const std::array<double, 3>& force,
+                         const double duration = 0.0);
+    bool applyWorldTorque(const std::array<double, 3>& torque,
+                          const double duration = 0.0);
+
+private:
+    class Impl;
+    std::unique_ptr<Impl> pImpl;
+};
+
+struct scenario::base::Pose
+{
+    Pose() = default;
+    Pose(std::array<double, 3> p, std::array<double, 4> o)
+        : position(p)
+        , orientation(o)
+    {}
+
+    // TODO: constructor
+    std::array<double, 3> position = {0, 0, 0};
+    std::array<double, 4> orientation = {1, 0, 0, 0};
```

Comment:
I understand that you don't want to document everything now, but I strongly suggest to briefly document the semantics of classes such as this or all the **velocity/acceleration** methods, as they are the one that is more easy to get wrong and introduce subtle bugs that invalidate the validity of a simulation. Even just a small comment pointing to the relative ignition math class with which they share the semantics is probably ok for now.

Reply:
Same answer https://github.com/robotology/gym-ignition/pull/158#discussion_r395504055